### PR TITLE
TPROD-275/TPROD-276 Refactor Event classes to Contracts and make dispatcher as first argument 

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR refactor the Event class to Symfony\Contracts\EventDispatcher\Event 
Also refactor by Rector plus two manually changes first argument event  for dispatcher

`$services->set(\Rector\Symfony\Rector\MethodCall\MakeDispatchFirstArgumentEventRector::class);`

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11140"></a>

### CODE REVIEW

**Comment by RCheesley** [comment]
Lots of tests failing here as well @kuzmany :(

File: app/bundles/LeadBundle/Field/Dispatcher/FieldSaveDispatcher.php [hunk 1]
```diff
@@ -46,40 +47,11 @@ public function dispatchPostSaveEvent(LeadField $entity, bool $isNew): LeadField
...
-    }
-
-    /**
-     * @param string $action - Use constant from LeadEvents class (e.g. LeadEvents::FIELD_PRE_SAVE)
+     * @param Event|null $event
      *
      * @throws NoListenerException
      */
-    private function dispatchEvent(string $action, LeadField $entity, bool $isNew, LeadFieldEvent $event = null): LeadFieldEvent
+    public function dispatchEvent(string $action, LeadField $entity, bool $isNew, $event = null): LeadFieldEvent
```
Comment (escopecz) [inline app/bundles/LeadBundle/Field/Dispatcher/FieldSaveDispatcher.php]: The `Event $event` type can be added back, right?

**Comment by biozshock** [comment]
@kuzmany , I've created a PR in your fork.

**Review by biozshock — APPROVED** [review]

**Review by biozshock — DISMISSED** [review]

**Review by escopecz — DISMISSED** [review]
I cannot find any problem with these changes. Thanks! 👍

**Review by escopecz — APPROVED** [review]

**Pull request merged**